src/allocators/AllocationScope.zig

```diff
@@ -216,6 +216,8 @@ pub fn trackExternalAllocation(scope: *AllocationScope, ptr: []const u8, ret_add
 /// Returns true if the free was invalid.
 pub fn trackExternalFree(scope: *AllocationScope, ptr: []const u8, ret_addr: ?usize) bool {
     if (!enabled) return;
+    // Empty slice usually means invalid pointer
+    if (ptr.len == 0) return false;
```

Comment:
Null-terminated slices must always be freed, and `ptr: []const u8` implicitly casts. So this is not correct.